wazuh/wazuh-dashboard-plugins
# Replace common components vuls module

### Description
This PR replaces and removes duplicated data grid, search bar, and doc viewer components/hooks.

Closes #6222 

## Tests

**Preconditions**
Add index `wazuh-states-vulnerabilities` with data following the [comment](https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103)

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

### Dashboard tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
|Check all the data is shown correctly (check all dashboard components are completed with data) in tops, metrics, table, etc) | :black_circle: | :black_circle: | :black_circle: |
| Click on add filter and check if the filter is applied | :black_circle: | :black_circle: | :black_circle: |
| Check dashboard table pagination works | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar input | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :black_circle: | :black_circle: | :black_circle: |
| Check dashboard responsive layout (mobile) | :black_circle: | :black_circle: | :black_circle: |

- Check all the data is shown correctly (check all dashboard components are completed with data) in tops, metrics, table, etc)

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/86418a3c-5d80-4f3b-a4f8-c48b14dbe464

</details>

- Click on add filter and check if the filter is applied

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/b18742bb-ab3d-4c5b-86ec-91cbf31329e2
</details>

- Check dashboard table pagination works

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/74c7437a-969d-4596-a717-445878f00c24
</details>

- Apply filters using the search bar input

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/c80dd1dc-5fa3-4ffc-b77b-8bf7d22f0deb
</details>

- Apply filters using the search bar `add filter` feature

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/3805e4b4-0525-4ff2-84fd-48f2789a5f2e
</details>

- Check dashboard responsive layout (vertical align in mobile)

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/075c0c12-e975-4de4-b1de-771b9d9648dd
</details>

### Inventory tab

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check all inventory table header features (sort, show/hide columns, density, etc) | :black_circle: | :black_circle: | :black_circle: |
| Check document detail flyout | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using search bar input | :black_circle: | :black_circle: | :black_circle: |
| Apply filters using the search bar `add filter` feature | :black_circle: | :black_circle: | :black_circle: |

- Check all inventory table header features (sort, show/hide columns, density, etc)

<details>

Screen Recording 2023-12-26 at 08.54.43
</details>

- Check document detail flyout

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/f766185b-19dd-4000-865d-55d07c107b05
</details>

- Apply filters using search bar input

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/9efedc99-ec56-46ec-a0bc-971efcf3cc66
</details>

- Apply filters using the search bar `add filter` feature

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/5f037bb6-063e-4204-91f7-885040cf8a7a
</details>

### Global

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check filters persistence between tabs (inventory, dashboard) | :black_circle: | :black_circle: | :black_circle: |
| Compare results between the inventory table and dashboard table | :black_circle: | :black_circle: | :black_circle: |

- Check filters persistence between tabs (inventory, dashboard)

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/5d1ed883-1b6d-4041-8092-367e6505071a
</details>

- Compare results between the inventory table and dashboard table

<details>

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/37a086ed-e6de-4154-ba5f-8a7fcaacdedf
</details>

### Check List
